### PR TITLE
Pgd lasso with callbacks

### PR DESCRIPTION
**Pull request by tanglef**
Example for [PR 137](https://github.com/benchopt/benchOpt/pull/137) with the callbacks.

- I changed the branch for the tests to go through (run them on the PR branch: to modify going further),
- Even if the `get_result` method is never used in `runer`, shouldn't it be better to keep it for consistency amongst the solvers (and avoid confusion for users)?

### CODE REVIEW

File: solvers/python_pgd_allinone.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+from benchopt import BaseSolver, safe_import_context
```
Comment (tomMoral) [inline solvers/python_pgd_allinone.py:1]: Maybe rename the file?

File: solvers/python_pgd_allinone.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        L = self.compute_lipschitz_cste()
+        n_features = self.X.shape[1]
+        w = np.zeros(n_features)
+        if self.use_acceleration:
+            z = np.zeros(n_features)
+
+        it = 0
+        t_new = 1
+        while callback(w):
+            it += 1
```
Comment (tomMoral) [inline solvers/python_pgd_allinone.py:71]:
```suggestion
```

File: solvers/python_pgd_allinone.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return rk
+
+    def run(self, callback):
+        L = self.compute_lipschitz_cste()
+        n_features = self.X.shape[1]
+        w = np.zeros(n_features)
+        if self.use_acceleration:
+            z = np.zeros(n_features)
+
+        it = 0
```
Comment (tomMoral) [inline solvers/python_pgd_allinone.py:68]:
```suggestion
```

**Comment by agramfort** [comment]
one more down !

**Pull request merged**